Unknown argument: uglify

PR #4755 introduced a ` --no-uglify` flag but it doesn't work for me when invoked from an npm script or via `npx`:

```
$ npx gatsby build --no-uglify
node_modules\gatsby\dist\bin\gatsby.js build

Options:
  -h, --help      Show help                                                       
  --verbose       Turn on verbose output                                          
  --no-color      Turn off the color in output                                    
  --prefix-paths  Build site with link paths prefixed (set prefix in your config).
  --no-uglify     Build site without uglifying JS bundles (for debugging).        
  -v, --version   Show version number                                             

Unknown argument: uglify
```

As a workaround, it is possible to do

```
$ npx gatsby build --noUglify
```

or

```
{
  "scripts": {
    "build": "gatsby build --noUglify"
  }
}
```

---

Note: I previously made it sound like the CLI parsing is incorrect but doing `$ gatsby build --no-uglify` (using global gatsby-cli) works correctly. Sorry!

I'm able to reproduce this on my mac if that helps:

```
$ yarn build
yarn run v1.6.0
$ gatsby build --no-uglify
gatsby build

Build a Gatsby project.

Options:
  --verbose       Turn on verbose output                                                        [boolean] [default: false]
  --no-color      Turn off the color in output                                                  [boolean] [default: false]
  --prefix-paths  Build site with link paths prefixed (set prefix in your config).              [boolean] [default: false]
  --no-uglify     Build site without uglifying JS bundles (for debugging).                      [boolean] [default: false]
  -h, --help      Show help                                                                                      [boolean]
  -v, --version   Show version number                                                                            [boolean]

Unknown argument: uglify
```

Does the same thing when run by NPM:

```
$ npm run build

> <project name>@0.1.0 build <project directory>
> gatsby build --no-uglify

gatsby build

Build a Gatsby project.

Options:
  --verbose       Turn on verbose output                                                        [boolean] [default: false]
  --no-color      Turn off the color in output                                                  [boolean] [default: false]
  --prefix-paths  Build site with link paths prefixed (set prefix in your config).              [boolean] [default: false]
  --no-uglify     Build site without uglifying JS bundles (for debugging).                      [boolean] [default: false]
  -h, --help      Show help                                                                                      [boolean]
  -v, --version   Show version number                                                                            [boolean]

Unknown argument: uglify
npm ERR! code ELIFECYCLE
npm ERR! errno 1

<...etc>
```

If there's any info I can give you guys about my setup I'm happy to.

--- a/packages/gatsby/package.json
+++ b/packages/gatsby/package.json
@@ -193,5 +193,8 @@
     "watch": "rimraf dist && mkdir dist && npm run build:internal-plugins && npm run build:rawfiles && npm run build:src -- --watch"
   },
   "types": "index.d.ts",
-  "gitHead": "788133e56dd005663f353fff90ffa37ce2fd87b9"
+  "gitHead": "788133e56dd005663f353fff90ffa37ce2fd87b9",
+  "yargs": {
+    "boolean-negation": false
+  }
 }

